### PR TITLE
Add calculateOptimalRotation

### PR DESCRIPTION
**Pull request by rakrakon**

### CODE REVIEW

File: pre-commit
Comment (katzuv) [inline pre-commit:1]: Shouldn't be changed here.

File: src/main/java/lib/Utils.java [hunk 1]
```diff
@@ -60,4 +60,11 @@ public static Pose2d calcOptimalPose(List<Pose2d> points, Pose2d robotPose) {
                                 point -> getDistanceFromPoint(point.getTranslation(), robotPose)))
                 .orElse(null);
     }
+
+    public static Rotation2d calculateOptimalRotation(
```
Comment (katzuv) [inline src/main/java/lib/Utils.java]: The name is a bit excessive, there is nothing "optimal" in this method.

File: src/main/java/frc/robot/swerve/SwerveDrive.java [hunk 2]
```diff
@@ -258,6 +260,23 @@ public Command driveCommand(
                                 fieldOriented.getAsBoolean()));
     }
 
+    public Command turnCommand(double rotation, double turnTolerance) {
+        PIDController turnController =
```
Comment (katzuv) [inline src/main/java/frc/robot/swerve/SwerveDrive.java:264]: Better to use a motion profile.

**Review by katzuv — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/swerve/SwerveDrive.java (hunk 2 above)
Comment (rakrakon) [reply src/main/java/frc/robot/swerve/SwerveDrive.java:264]: Eitan said Dieter controller works best for rotations

File: src/main/java/lib/Utils.java (hunk 1 above)
Comment (rakrakon) [reply src/main/java/lib/Utils.java:64]: ca4e6a4

File: pre-commit
Comment (rakrakon) [reply pre-commit:1]: ca4e6a4

File: src/main/java/lib/Utils.java (hunk 1 above)
Comment (rakrakon) [reply src/main/java/lib/Utils.java]: 74eea28

**Review by katzuv — APPROVED** [review]

**Pull request merged**